### PR TITLE
perf(token): update token per new theming #21

### PR DESCRIPTION
**Pull request by irma-kurnia-alaska**
# Alaska Airlines Pull Request

**Fixes:** #21 

## Summary:

- Update the design tokens to reflect the new theming conventions regarding tokens from the AuroDesignTokens and JetstreamDesignTokens repo.

## Type of change:

Please delete options that are not relevant.

- [x] Revision of an existing capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Let's include the switch from node-sass to sass in this PR, please.

https://dev.to/ccreusat/migrating-from-node-sass-to-sass-dart-sass-with-npm-3g3c

**Comment by irma-kurnia-alaska** [comment]
@jason-capsule42,

I will fix as your feedback, please reviewed. Thank you.

File: package.json [hunk 1]
```diff
@@ -107,11 +110,12 @@
...
     "ciBuild": "npm-run-all sassBuild sassRender cssLint distJS bundler postinstall apiBuild",
-    "sassBuild": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:dev": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:demo": "node-sass ./demo/sass/ --output ./demo/css/",
-    "sassBuild:component": "node-sass ./src/ --output ./src/",
+    "sassBuild": "npm-run-all sassBuild:component postCss:component sassRender",
+    "sassBuild:dev": "npm-run-all sassBuild:component postCss:component sassRender",
+    "sassBuild:component": "sass --no-source-map src:src",
     "sassBuild:watch": "nodemon -e scss --watch src --exec npm run sassBuild:dev",
+    "sassBuild:debug": "sass —source-map scss/index.scss docs/index.css",
```
Comment (blackfalcon) [inline package.json:117]: Why was this added? I as configured I am not able to get this to work? 

``` npm ERR! code ELIFECYCLE
npm ERR! errno 64
npm ERR! @alaskaairux/auro-loader@1.1.1 sassBuild:debug: `sass —source-map scss/index.scss docs/index.css`
npm ERR! Exit status 64
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -80,7 +82,8 @@
     "wc-sass-render": "^1.2.3",
     "web-component-analyzer": "^1.1.6",
     "webpack-merge": "^5.4.0",
-    "whatwg-fetch": "^3.5.0"
+    "whatwg-fetch": "^3.5.0",
+    "sass": "^1.49.9"
```
[the remark this answers is not in the document]
Comment (irma-kurnia-alaska) [reply package.json:86]: ok, Thank you.

File: package.json (hunk 1 above)
Comment (irma-kurnia-alaska) [reply package.json:117]: I followed the url to added the code, https://dev.to/ccreusat/migrating-from-node-sass-to-sass-dart-sass-with-npm-3g3c, i think it's my mistake sorry i will remove and fix as you suggest.

File: package.json [hunk 3]
```diff
@@ -107,11 +110,12 @@
     "test:watch": "karma start --auto-watch=true --single-run=false",
     "test:ci": "npm-run-all test lint",
     "ciBuild": "npm-run-all sassBuild sassRender cssLint distJS bundler postinstall apiBuild",
-    "sassBuild": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:dev": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:demo": "node-sass ./demo/sass/ --output ./demo/css/",
```
[the remark this answers is not in the document]
Comment (irma-kurnia-alaska) [reply package.json:112]: Hi @blackfalcon,

I will be follow this https://dev.to/ccreusat/migrating-from-node-sass-to-sass-dart-sass-with-npm-3g3c, but i think i wrong, i'm sorry for this. I will fix again as you suggestion. thank you.

[the remark this answers is not in the document]
Comment (irma-kurnia-alaska) [reply package.json:112]: yes , @blackfalcon , so I revert to the lates code and I fix as your suggestion. Thank you @blackfalcon.

File: package.json [hunk 4]
```diff
@@ -73,6 +73,7 @@
     "rollup-plugin-minify-html-literals": "^1.2.5",
     "rollup-plugin-serve": "^1.1.0",
     "rollup-plugin-terser": "^7.0.2",
+    "sass": "^1.49.9",
```
Comment (jason-capsule42) [inline package.json:76]: After adding the sass tool you should be able to remove this from the devDependenies.

"wc-sass-render": "^1.2.3",

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Review by blackfalcon — APPROVED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please update the testPublish.yml file to the following:

```yml
name: Test and publish

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the main branch
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [12.x, 14.x]

    steps:
      - uses: actions/checkout@v3
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
      - run: npm ci
      - run: npm run sassBuild:component
      - run: npm run postCss:component
      - run: npm run sassRender
      - run: npm run distJS
      - run: npm run bundler
      - run: npm test

  release:
    # Only release on push to main
    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
    runs-on: ubuntu-latest
    needs: test
    steps:
      - uses: actions/checkout@v3
        with:
          persist-credentials: false
      - uses: actions/setup-node@v3
        with:
          node-version: 14
      - run: npm ci
      - run: npm run sassBuild:component
      - run: npm run postCss:component
      - run: npm run sassRender
      - run: npm run distJS
      - run: npm run bundler
      - uses: cycjimmy/semantic-release-action@v3
        env:
          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

File: src/styles/_options.scss [hunk 5]
```diff
@@ -1,11 +1,11 @@
 :host([onlight]) {
-  color: var(--auro-color-ui-default-on-light);
+  color: var(--ds-color-ui-default-default, $ds-color-ui-default-default);
 }
 
 :host([ondark]) {
-  color: var(--auro-color-ui-default-on-dark);
+  color: var(--ds-color-brand-breeze-300, $ds-color-brand-breeze-300);
```
[the remark this answers is not in the document]
Comment (leeejune) [reply src/styles/_options.scss:6]: Yes! The correct "replacement" should be `var(--ds-color-ui-default-inverse)`

[the remark this answers is not in the document]
Comment (jason-capsule42) [reply src/styles/_options.scss:6]: @irma-kurnia-alaska can you make this change?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
See comments

File: demo/sass/style.scss [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
 $layout-modifiers: (
-  "md": $auro-size-md
+  "md": $ds-size-200
 );
-@import "./node_modules/@alaskaairux/webcorestylesheets/dist/utilityMixins/layoutPropertiesGenerator";
+
+@import "../../node_modules/@aurodesignsystem/webcorestylesheets/src/utilityMixins/layoutPropertiesGenerator";
 
 // Custom styles per demo pages
 code {
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/sass/style.scss:23]: @blackfalcon Any update on this? There is no `--auro-color-base-crimson-500` in the deprecated tokens list.

**Review by jordanjones243 — APPROVED** [review]

**Comment by blackfalcon** [comment]
Closing this PR as work was integrated into the repo upgrade project.
